k8sapiserver/k8sapiserver.go

```diff
@@ -130,6 +130,8 @@ func NewControlPlaneConfigWithOptions(serverURL, etcdURL string) *controlplane.C
 
 	genericConfig.SecureServing = &genericapiserver.SecureServingInfo{Listener: fakeLocalhost443Listener{}}
 
+	genericConfig.CorsAllowedOriginList = []string{"http://*"}
```

Comment:
Hi @momom-i!
Cc. @sanposhiho 

From a perspective of security, you should indicate the frontend url like this, I think.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/fc420bc2e9e55fedc13adba583ee41c9020387fa/server/server.go#L31
(And current code  maybe not  allow a request from `HTTPS`.)

But this `genericConfig` is also applied to `ectd`. Unfortunately, the above configuration is for the `API server`. (The `etcd` is accessed by only`API server`.)
 Therefore , you should separate that configuration logic I think.

Reply:
> But this genericConfig is also applied to ectd. Unfortunately, the above configuration is for the API server. (The etcd is accessed by onlyAPI server.)

@196Ikuchil Is there any documentation for this? I did not know about this. (Either way, I think @196Ikuchil's idea is valid :) )

Reply:
> But this genericConfig is also applied to ectd.

The point I was trying to make is that `genericConfig` is used in both 
[etcdOptions.ApplyWithStorageFactoryTo...](https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/115/files/d63a12c1166b215ac156f1e7c05b774334d329a7#diff-7dc54cbad11a8158cbb340b7757b85aef66049765a514b15908cf940a6d44185R134) and [cfg := &controlplane.Config{GenericConfig: genericConfig,...](https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/115/files/d63a12c1166b215ac156f1e7c05b774334d329a7#diff-7dc54cbad11a8158cbb340b7757b85aef66049765a514b15908cf940a6d44185R140)
And I think it is not good to apply that `AllowOrigins` configuration to `etcd`.

Therefore, there is no documentation, sorry.

Reply:
I had a quick look at the implementation of `ApplyWithStorageFactoryTo`. But I don't think the AllowOrigins configuration is applying to etcd. (In the first place, there is no `AllowOrigins` field in the etcd options.)

@momom-i 
So, using `genericConfig. CorsAllowedOriginList` looks good to me, please change `"http://*"` to use `cfg.FrontendURL` like the simulator server.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/fc420bc2e9e55fedc13adba583ee41c9020387fa/server/server.go#L31

Reply:
I'm sorry for my late reply. Thank you for your comment, @196Ikuchil and also @sanposhiho !
I've confirmed `EtcdOptions` has no `AllowOrigins` field,  so I'll change like what you said.